pkg/investigations/insightsoperatordown/insightsoperatordown.go

```diff
@@ -0,0 +1,133 @@
+	err = k8scli.List(context.TODO(), coList, listOptions)
+	if err != nil {
+		return result, fmt.Errorf("unable to list insights clusteroperator: %w", err)
+	}
+
+	if len(coList.Items) != 1 {
+		return result, fmt.Errorf("found %d clusteroperators, expected 1", len(coList.Items))
+	}
+	co := coList.Items[0]
+
+	if isOCPBUG22226(&co) {
+		notes.AppendWarning("Found symptom of OCPBUG22226. Try deleting the insights operator pod to remediate.\n$ oc -n openshift-insights delete pods -l app=insights-operator --wait=false")
+		return result, r.PdClient.EscalateIncidentWithNote(notes.String())
+	} else {
+		notes.AppendSuccess("Ruled out OCPBUGS-22226")
+	}
+
+	verifierResult, failureReason, err := networkverifier.Run(r.Cluster, r.ClusterDeployment, r.AwsClient)
+	if err != nil {
+		logging.Error("Network verifier ran into an error: %s", err.Error())
+		notes.AppendWarning("NetworkVerifier failed to run:\n\t %s", err.Error())
+
+		err = r.PdClient.AddNote(notes.String())
+		if err != nil {
+			// We do not return as we want the alert to be escalated either no matter what.
+			logging.Error("could not add failure reason incident notes")
+		}
+	}
+
+	switch verifierResult {
+	case networkverifier.Failure:
+		logging.Infof("Network verifier reported failure: %s", failureReason)
+		// XXX: metrics.Inc(metrics.ServicelogPrepared, investigationName)
+		result.ServiceLogPrepared = investigation.InvestigationStep{Performed: true, Labels: nil}
+		notes.AppendWarning("NetworkVerifier found unreachable targets. \n \n Verify and send service log if necessary: \n osdctl servicelog post %s -t https://raw.githubusercontent.com/openshift/managed-notifications/master/osd/required_network_egresses_are_blocked.json -p URLS=%s", r.Cluster.ID(), failureReason)
+
+		// In the future, we want to send a service log in this case
+		err = r.PdClient.AddNote(notes.String())
+		if err != nil {
+			logging.Error("could not add issues to incident notes")
+		}
+	case networkverifier.Success:
+		notes.AppendSuccess("Network verifier passed")
+		err = r.PdClient.AddNote(notes.String())
+		if err != nil {
+			logging.Error("could not add passed message to incident notes")
+		}
+	}
+	return result, r.PdClient.EscalateIncidentWithNote(notes.String())
+}
+
+func isOCPBUG22226(co *configv1.ClusterOperator) bool {
+	symptomStatusString := "Failed to pull SCA certs"
+
+	for _, condition := range co.Status.Conditions {
+		if condition.Type == "SCAAvailable" && strings.Contains(condition.Message, symptomStatusString) {
+			return true
+		}
+	}
+	return false
+}
+
+func (c *Investigation) Name() string {
+	return "insightsoperatordown"
+}
+
+func (c *Investigation) Description() string {
+	return "Investigate insights operator down alert"
+}
+
+func (c *Investigation) ShouldInvestigateAlert(alert string) bool {
+	return strings.Contains(alert, "InsightsOperatorDown")
+}
+
+func (c *Investigation) IsExperimental() bool {
+	return true
```

Comment:
This is informational already, we can put this to false to have it run on prod imo.